### PR TITLE
Master social login

### PR DESCRIPTION
**Pull request by RafaFP**
Adding Facebook and Google authentication Login to Mconf login

### CODE REVIEW

File: config/locales/en/mconf.yml [hunk 1]
```diff
@@ -689,6 +689,10 @@ en:
         other_options: "Other options"
         register: "Create a new account"
         resend_confirmation_email: "Resend confirmation email"
+      social_login:
+        social_facebook: "Login with Facebook Account"
```
Comment (daronco) [inline config/locales/en/mconf.yml]: "Log in with Facebook"

File: config/locales/en/mconf.yml [hunk 2]
```diff
@@ -689,6 +689,10 @@ en:
...
+        social_google: "Login with Google Account"
```
Comment (daronco) [inline config/locales/en/mconf.yml]: "Log in with Google"

File: config/locales/pt-br/mconf.yml [hunk 3]
```diff
@@ -697,6 +697,10 @@ pt-br:
         other_options: "Outras opções"
         register: "Criar uma nova conta"
         resend_confirmation_email: "Reenviar e-mail de confirmação"
+      social_login:
+        social_facebook: "Acessar com a conta Facebook"
```
Comment (daronco) [inline config/locales/pt-br/mconf.yml]: "Entrar com o Facebook"

File: config/locales/pt-br/mconf.yml [hunk 4]
```diff
@@ -697,6 +697,10 @@ pt-br:
...
+        social_google: "Acessar com a conta Google"
```
Comment (daronco) [inline config/locales/pt-br/mconf.yml]: "Entrar com o Google"

File: .env.development [hunk 5]
```diff
@@ -28,3 +28,11 @@ MCONF_EMAIL_TRACK_CLICKED=true
 # MCONF_REDIS_HOST='localhost'
 # MCONF_REDIS_PORT='6379'
 # MCONF_REDIS_DB='0'
+
+# Omniauth for google and facebook need API keys.
+# More instructions can be found on https://github.com/omniauth/omniauth/wiki/List-of-Strategies
+# Then add the following:
+# MCONF_OMNIAUTH_GOOGLE_KEY = 'yourgooglekey'
```
[the remark this answers is not in the document]
Comment (RafaFP) [reply .env.development]: Fixed that

File: app/models/user.rb [hunk 6]
```diff
@@ -165,6 +165,22 @@ def self.find_first_by_auth_conditions(warden_conditions)
   # set to true when the user signs in via an external authentication method (e.g. LDAP)
   attr_accessor :signed_in_via_external
 
+  # define the creation for social omniauth
+  def self.from_omniauth(access_token)
+    data = access_token.info
+    user = User.where(email: data['email']).first
```
[the remark this answers is not in the document]
Comment (RafaFP) [reply app/models/user.rb]: Completely forgot I could do that. Fixed.

File: app/views/devise/shared/_social_login.html.haml [hunk 7]
```diff
@@ -0,0 +1,18 @@
+#social-login-section
+  - if (Rails.application.config.omniauth_google_key.present? && Rails.application.config.omniauth_google_secret.present?) || (Rails.application.config.omniauth_facebook_key.present? && Rails.application.config.omniauth_facebook_secret.present?)
```
[the remark this answers is not in the document]
Comment (RafaFP) [reply app/views/devise/shared/_social_login.html.haml]: Done. Looks much better now

File: config/locales/en/mconf.yml (hunk 1 above)
Comment (RafaFP) [reply config/locales/en/mconf.yml]: Done

File: config/locales/en/mconf.yml (hunk 2 above)
Comment (RafaFP) [reply config/locales/en/mconf.yml]: Done

File: config/locales/pt-br/mconf.yml (hunk 4 above)
Comment (RafaFP) [reply config/locales/pt-br/mconf.yml]: Done

File: config/locales/pt-br/mconf.yml (hunk 3 above)
Comment (RafaFP) [reply config/locales/pt-br/mconf.yml]: Done

File: .env.test [hunk 8]
```diff
@@ -5,3 +5,11 @@ MCONF_DATABASE_TEST_PASSWORD=''
 # otherwise AhoyEmail will change links and break tests
 MCONF_EMAIL_TRACK_CLICKED=false
 MCONF_EMAIL_TRACK_OPENED=true
+
```
[the remark this answers is not in the document]
Comment (RafaFP) [reply .env.test]: No, we actually don't, they are stubbed for testing.

[the remark this answers is not in the document]
Comment (RafaFP) [reply .env.test]: I am using this on a feature test scenario

[the remark this answers is not in the document]
Comment (RafaFP) [reply .env.test]: Yes. We need them so that devise can initialize the omniauth

File: app/models/user.rb [hunk 9]
```diff
@@ -168,7 +168,7 @@ def self.find_first_by_auth_conditions(warden_conditions)
   # define the creation for social omniauth
   def self.from_omniauth(access_token)
     data = access_token.info
-    user = User.where(email: data['email']).first
+    user = User.find_by_email(data['email'])
```
Comment (daronco) [inline app/models/user.rb]: This is an old syntax, it's better to use `find_by(email: data['email'])`.

File: app/views/devise/shared/_social_login.html.haml [hunk 10]
```diff
@@ -1,15 +1,16 @@
 #social-login-section
-  - if (Rails.application.config.omniauth_google_key.present? && Rails.application.config.omniauth_google_secret.present?) || (Rails.application.config.omniauth_facebook_key.present? && Rails.application.config.omniauth_facebook_secret.present?)
+  - if Site.current.social_login_enabled?
```
Comment (daronco) [inline app/views/devise/shared/_social_login.html.haml]: We have a [`current_site`](https://github.com/mconf/mconf-web/blob/b7095825ee2fbccffae0d38b58b2e84e7196f1da/app/controllers/application_controller.rb#L58) helper that caches the site, it's usually better to use it in views.

**Pull request merged**